setup.py

```diff
@@ -224,66 +95,114 @@ def get_git_changeset():
-        'beancount.ingest.importers',
-        'beancount.ingest.importers.mixins',
-        'beancount.utils',
-        'beancount.tools',
-    ],
-
-    package_data = {
-        'beancount.web': ['*.ico',
-                          '*.html',
-                          '*.css',
-                          'third_party/*.js'],
-        'beancount.reports': ['*.html'],
-        'beancount.utils.file_type_testdata': ['*'],
-        'beancount.parser': ['*.h'], # See note for {63fc8d84d30a} above.
-        },
-
-    ext_modules = [
-        Extension("beancount.parser._parser",
-                  sources=[
-                      "beancount/parser/lexer.c",
-                      "beancount/parser/grammar.c",
-                      "beancount/parser/parser.c",
-                  ],
-                  define_macros=[
-                      ('RELEASE_VERSION', version),
-                      ('VC_CHANGESET', vc_changeset),
-                      ('VC_TIMESTAMP', int(float(vc_timestamp))),
-                      ('PARSER_SOURCE_HASH', hash_parser_source_files())],
-                  extra_compile_args=get_cflags()),
-    ],
-
-    # Include the Emacs support for completeness, for packagers not to have to
-    # check out from the repository.
-    data_files = [
-        ('elisp', ['editors/emacs/beancount.el']),
-    ],
-
-    # Add optional arguments that only work with some variants of setup().
-    **setup_extra_kwargs
+      """,
+      
+      license="GNU GPLv2 only",
+      author="Martin Blais",
+      author_email="blais@furius.ca",
+      url="http://furius.ca/beancount",
+      download_url="https://github.com/beancount/beancount",
+      packages=find_packages(exclude=['experiments*', ]),
+
+      package_data = {
+          'beancount.web': ['*.ico',
+                            '*.html',
+                            '*.css',
+                            'third_party/*.js'],
+          'beancount.reports': ['*.html'],
+          'beancount.utils.file_type_testdata': ['*'],
+      },
+
+      ext_modules = [
+          Extension("beancount.parser._parser",
+                    sources=[
+                        "beancount/parser/lexer.c",
+                        "beancount/parser/grammar.c",
+                        "beancount/parser/parser.c",
+                    ],
+                    define_macros=[
+                        ('RELEASE_VERSION', version),
+                        ('VC_CHANGESET', vc_changeset),
+                        ('VC_TIMESTAMP', int(float(vc_timestamp))),
+                        ('PARSER_SOURCE_HASH', hash_parser_source_files())],
+                extra_compile_args=get_cflags()),
+      ],
+
+      # Include the Emacs support for completeness, for packagers not to have to
+      # check out from the repository.
+      data_files = [
+          ('elisp', ['editors/emacs/beancount.el']),
+      ],
+
+      install_requires = [
+          # Testing support now uses the pytest module.
+          'pytest',
```

Comment:
Could we move pytest to development requirements? As in this SO answer https://stackoverflow.com/a/28842733/1868395

Reply:
+1

Reply:
Sure, this was one of the reasons for reworking `setup.py`. However, this is a behavior change and I was thinking to do it in a separate pull request. With @blais blessing it, I'll add it to this one.